lib/src/model/language_feature.dart

```diff
@@ -35,13 +33,36 @@ class LanguageFeature {
   String? get featureUrl => _featureUrls[name];
 
   /// The rendered label for this language feature.
-  String get featureLabel => _featureRenderer.renderLanguageFeatureLabel(this);
+  String get featureLabel {
+    final buffer = StringBuffer();
+    final url = featureUrl;
+
+    if (url != null) {
+      buffer.write('<a href="');
+      buffer.write(url);
+      buffer.write('"');
+    } else {
+      buffer.write('<span');
+    }
+
+    buffer.write(' class="feature feature-');
+    buffer.writeAll(name.toLowerCase().split(' '), '-');
+    buffer.write('" title="');
+    buffer.write(featureDescription);
+    buffer.write('">');
+    buffer.write(name);
+
+    if (url != null) {
+      buffer.write('</a>');
+    } else {
+      buffer.write('</span>');
+    }
+
+    return buffer.toString();
+  }
 
   /// The name of this language feature.
   final String name;
 
-  final LanguageFeatureRenderer _featureRenderer;
-
-  LanguageFeature(this.name, this._featureRenderer)
-      : assert(_featureDescriptions.containsKey(name));
+  LanguageFeature(this.name) : assert(_featureDescriptions.containsKey(name));
```

Comment:
Not very important, but in a different train of thought, I'm not even sure if it should be called `LanguageFeature`. Are these not just class modifiers? Did we have more plans for these? https://github.com/dart-lang/dartdoc/issues/3400

Reply:
:kermit-the-frog-scrunch-face:

I don't know. Very strange. But good find with https://github.com/dart-lang/dartdoc/issues/3400! Yeah this could be renamed. And easier with everything being private API now.